AC-1394 Review Glue API backend docs (#1269)
spryker/spryker-docs: docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-create-resources-with-parent-child-relationships.md

---
title: How to create resources with parent-child relationships
description: This document shows how to create resources with parent-child relationships 
last_updated: September 30, 2022
template: howto-guide-template
redirect_from:
  - /docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/create-a-resource-with-parent.html
---

Glue API lets you create resources with parent-child relationships or, in other words, nested resources. In order to enable such behavior, a resource that implements `Spryker\Glue\GlueApplicationExtension\Dependency\Plugin\ResourceWithParentPluginInterface` must be created.

Such a plugin routes requests from the parent resources to the correct child resource. This interface must be implemented together with `ResourceInterface`.

The interface provides only one method: `getParentResourceType`. The method must return the type of the immediate parent resource within the context in which the child resource is implemented.

> Let's say you have a module named `FooApi`, where you want to have a new endpoint `/foo/1/bar` with `GET` and `POST` methods. For this, follow these steps:

Author: ```suggestion
Let's say you have a module named `FooApi`, where you want to have a new endpoint `/foo/1/bar` with `GET` and `POST` methods. To create the new endpoint, follow these steps:
```

> 1. Create a resource following the how to create a resource guide.

Author: ```suggestion
1. Create a resource following the [how to create a resource](/docs/scos/dev/glue-api-guides/{{page.version}}\glue-backend-api/how-to-guides/how-to-create-a-resource.html) guide.
```

}
```

> If everything is set up correctly, you can access `https://glue-storefront.mysprykershop.com/foo/1/bar`.

Author: We must avoid using "should" - https://developers.google.com/style/word-list#should
```suggestion
Now, you can access `https://glue-storefront.mysprykershop.com/foo/1/bar`.
```
